### ISSUE TITLE
Move Typescript definitions from definitely-typed into this package

### ISSUE THREAD

**Issue by ncphillips** [react-dropzone/react-dropzone#404]
**Do you want to request a *feature* or report a *bug*?**

- [ ] I found a bug
- [X] I want to propose a feature

**What is the current behavior?**
Must install `@types/react-dropzone` to your project to use with typings

**What is the expected behavior?**
Include a `typings.d.ts` file directly in this project

**If this is a feature request, what is motivation or use case for changing the behavior?**

* Fewer dependencies
* Nice to keep definitions close to source code
* Projects using `react-dropzone` don't have to worry about definitions not matching source.
* Versionin number makes more sense. i.e. current version of `@types/react-dropzone` is `0.0.32` and hasn't been updated in 2 months.
* Easier for me to contribute to source then to definitely typed.
* Remembering to keep up changes

**Comment by ncphillips** [comment]
I can definitely help with this. Just waiting on the go-ahead.

**Comment by okonet** [comment]
Would be happy to accept a PR!

**Comment by ncphillips** [comment]
I believe @andy-ms originally created that typing

**Comment by andy-ms** [comment]
Nope, it was created by @matdube, @LynxEyes, and @goblindegook, all I did was move the file.

**Comment by okonet** [comment]
Closing of inacivity

**Comment by jeremejevs** [comment]
Why not keep open and mark with `help-wanted` instead?

**Comment by okonet** [comment]
Good idea

**Comment by patrick91** [comment]
I might work on this this weekend, I'm using react dropzone in TS project :)

**Comment by okonet** [comment]
It would be great to have something automated on CI, though.

**Comment by patrick91** [comment]
@okonet what do you mean?

**Comment by okonet** [comment]
I mean it would be amazing to generate and publish type definitions when the new version is published. I imaging this should be doable on CI step but I'm open for options.

**Comment by patrick91** [comment]
@okonet ah I see, well once we have the typings in this repo there's no need to publish another package for them, so when a user installs `react-dropzone` they will receive the latest typings automatically (since they will come with the npm package). Hope it makes sense[1]

We did this on Styled Components, see here: https://github.com/styled-components/styled-components/blob/master/package.json#L5

[1] This probably makes more sense :) https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html

**Comment by okonet** [comment]
Oh yeah it would be even better. Would be nice if you could work on it.

**Comment by jeremejevs** [comment]
It's actually probably best to keep the typings in DefinitelyTyped after all. See this discussion in `redux-thunk` for some pros and cons: https://github.com/reduxjs/redux-thunk/pull/180#issuecomment-387660405 Long story short, unless the target project is written in TypeScript itself, cons of co-location outweigh the pros.

**Comment by okonet** [comment]
Hmm. Ideally we would provide types for both flow and TS. We have a branch with flow typings for the whole project. How hard is it to keep TS definition in alignment if we’d switch to Flow?

**Comment by jeremejevs** [comment]
Honestly, don't know. The typings themselves are pretty simple, almost copy-paste of Flow, but then there are also tests. Why do you want to provide types? As mentioned in that tweet, DefinitelyTyped has a lot going for it: https://twitter.com/drosenwasser/status/981722403033309185

**Comment by okonet** [comment]
Anyone wants to work on this?
